### ISSUE TITLE
[ERROR] Failed to execute goal org.apache.thrift.tools:maven-thrift-plugin:0.1.11:compile (thrift-sources) on project datahandler: thrift did not exit cleanly

### ISSUE THREAD

**Issue by dineshr93** [eclipse/sw360#807]
I followed the instruction in [Deploy-docker](https://github.com/eclipse/sw360/wiki/Deploy-Docker)
on machine running Ubuntu 16.04

```
dira@server:/usr/local/src/sw360$ thrift --version
Thrift version 0.9.1
```

Maven build is failing 

```
dira@server:/usr/local/src/sw360$ sudo mvn package -P deploy -Dbase.deploy.dir=/usr/local/src/sw360chores/_deploy -DskipTests -Dmaven.repo.local=/usr/local/src/mavenrepo -e
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] build-configuration
[INFO] datahandler
[INFO] commonIO
[INFO] exporters
[INFO] src-common
[INFO] src-attachments
[INFO] src-components
[INFO] src-vendors
[INFO] importers
[INFO] libraries
[INFO] src-licenses
[INFO] src-users
[INFO] src-search
[INFO] src-projects
[INFO] src-moderation
[INFO] src-fossology
[INFO] src-vulnerabilities
[INFO] src-licenseinfo
[INFO] src-cvesearch
[INFO] src-schedule
[INFO] src-wsimport
[INFO] backend-src
[INFO] svc-common
[INFO] svc-licenses
[INFO] svc-users
[INFO] svc-vendors
[INFO] svc-search
[INFO] svc-components
[INFO] svc-projects
[INFO] svc-attachments
[INFO] svc-moderation
[INFO] svc-fossology
[INFO] svc-vulnerabilities
[INFO] svc-cvesearch
[INFO] svc-licenseinfo
[INFO] svc-schedule
[INFO] svc-wsimport
[INFO] backend-svc
[INFO] backend-utils
[INFO] backend
[INFO] sw360-portlet
[INFO] liferay-theme
[INFO] frontend
[INFO] rest-common
[INFO] authorization-server
[INFO] resource-server
[INFO] rest
[INFO] sw360
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building build-configuration 8.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- buildnumber-maven-plugin:1.4:create (default) @ build-configuration ---
[INFO] ShortRevision tag detected. The value is '7'.
[INFO] Executing: /bin/sh -c cd '/usr/local/src/sw360/build-configuration' && 'git' 'rev-parse' '--verify' '--short=7' 'HEAD'
[INFO] Working directory: /usr/local/src/sw360/build-configuration
[INFO] Storing buildNumber: 0882b8d at timestamp: 1583934450984
[INFO] Storing buildScmBranch: master
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce) @ build-configuration ---
[INFO]
[INFO] --- jacoco-maven-plugin:0.7.2.201409121644:prepare-agent (default) @ build-configuration ---
[INFO] argLine set to -javaagent:/usr/local/src/mavenrepo/org/jacoco/org.jacoco.agent/0.7.2.201409121644/org.jacoco.agent-0.7.2.201409121644-runtime.jar=destfile=/usr/local/src/sw360/build-configuration/target/jacoco.exec,excludes=org/xmlsoap/schemas/**:org/apache/ws/commons/schema/** -Dcatalina.home=/usr/local/src/sw360/build-configuration/target/home -Xms64m -Xmx128m -XX:MaxPermSize=512m
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ build-configuration ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 4 resources
[INFO]
[INFO] --- flatten-maven-plugin:1.0.0:flatten (flatten) @ build-configuration ---
[INFO] Generating flattened POM of project org.eclipse.sw360:build-configuration:jar:8.1.0-SNAPSHOT...
[INFO]
[INFO] --- maven-compiler-plugin:3.1:compile (default-compile) @ build-configuration ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ build-configuration ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 1 resource
[INFO]
[INFO] --- maven-compiler-plugin:3.1:testCompile (default-testCompile) @ build-configuration ---
[INFO] No sources to compile
[INFO]
[INFO] --- maven-surefire-plugin:2.17:test (default-test) @ build-configuration ---
[INFO] Tests are skipped.
[INFO]
[INFO] --- jacoco-maven-plugin:0.7.2.201409121644:report (report) @ build-configuration ---
[INFO] Skipping JaCoCo execution due to missing execution data file:/usr/local/src/sw360/build-configuration/target/jacoco.exec
[INFO]
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ build-configuration ---
[INFO] Building jar: /usr/local/src/sw360/build-configuration/target/build-configuration-8.1.0-SNAPSHOT.jar
[INFO]
[INFO] --- maven-jar-plugin:2.4:test-jar (default) @ build-configuration ---
[INFO] Building jar: /usr/local/src/sw360/build-configuration/target/build-configuration-8.1.0-SNAPSHOT-tests.jar
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building datahandler 8.1.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- buildnumber-maven-plugin:1.4:create (default) @ datahandler ---
[INFO] ShortRevision tag detected. The value is '7'.
[INFO] Executing: /bin/sh -c cd '/usr/local/src/sw360/libraries/lib-datahandler' && 'git' 'rev-parse' '--verify' '--short=7' 'HEAD'
[INFO] Working directory: /usr/local/src/sw360/libraries/lib-datahandler
[INFO] Storing buildNumber: 0882b8d at timestamp: 1583934451949
[INFO] Storing buildScmBranch: master
[INFO]
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce) @ datahandler ---
[INFO]
[INFO] --- jacoco-maven-plugin:0.7.2.201409121644:prepare-agent (default) @ datahandler ---
[INFO] argLine set to -javaagent:/usr/local/src/mavenrepo/org/jacoco/org.jacoco.agent/0.7.2.201409121644/org.jacoco.agent-0.7.2.201409121644-runtime.jar=destfile=/usr/local/src/sw360/libraries/lib-datahandler/target/jacoco.exec,excludes=org/xmlsoap/schemas/**:org/apache/ws/commons/schema/** -Dcatalina.home=/usr/local/src/sw360/libraries/lib-datahandler/target/home -Xms64m -Xmx128m -XX:MaxPermSize=512m
[INFO]
[INFO] --- maven-thrift-plugin:0.1.11:compile (thrift-sources) @ datahandler ---
[ERROR] thrift failed output:
[ERROR] thrift failed error: [ERROR:/usr/local/src/sw360/libraries/lib-datahandler/src/main/thrift/users.thrift:58] (last token was 'RestApiToken')
Type "RestApiToken" has not been defined.

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] build-configuration ................................ SUCCESS [  1.277 s]
[INFO] datahandler ........................................ FAILURE [  0.126 s]
[INFO] commonIO ........................................... SKIPPED
[INFO] exporters .......................................... SKIPPED
[INFO] src-common ......................................... SKIPPED
[INFO] src-attachments .................................... SKIPPED
[INFO] src-components ..................................... SKIPPED
[INFO] src-vendors ........................................ SKIPPED
[INFO] importers .......................................... SKIPPED
[INFO] libraries .......................................... SKIPPED
[INFO] src-licenses ....................................... SKIPPED
[INFO] src-users .......................................... SKIPPED
[INFO] src-search ......................................... SKIPPED
[INFO] src-projects ....................................... SKIPPED
[INFO] src-moderation ..................................... SKIPPED
[INFO] src-fossology ...................................... SKIPPED
[INFO] src-vulnerabilities ................................ SKIPPED
[INFO] src-licenseinfo .................................... SKIPPED
[INFO] src-cvesearch ...................................... SKIPPED
[INFO] src-schedule ....................................... SKIPPED
[INFO] src-wsimport ....................................... SKIPPED
[INFO] backend-src ........................................ SKIPPED
[INFO] svc-common ......................................... SKIPPED
[INFO] svc-licenses ....................................... SKIPPED
[INFO] svc-users .......................................... SKIPPED
[INFO] svc-vendors ........................................ SKIPPED
[INFO] svc-search ......................................... SKIPPED
[INFO] svc-components ..................................... SKIPPED
[INFO] svc-projects ....................................... SKIPPED
[INFO] svc-attachments .................................... SKIPPED
[INFO] svc-moderation ..................................... SKIPPED
[INFO] svc-fossology ...................................... SKIPPED
[INFO] svc-vulnerabilities ................................ SKIPPED
[INFO] svc-cvesearch ...................................... SKIPPED
[INFO] svc-licenseinfo .................................... SKIPPED
[INFO] svc-schedule ....................................... SKIPPED
[INFO] svc-wsimport ....................................... SKIPPED
[INFO] backend-svc ........................................ SKIPPED
[INFO] backend-utils ...................................... SKIPPED
[INFO] backend ............................................ SKIPPED
[INFO] sw360-portlet ...................................... SKIPPED
[INFO] liferay-theme ...................................... SKIPPED
[INFO] frontend ........................................... SKIPPED
[INFO] rest-common ........................................ SKIPPED
[INFO] authorization-server ............................... SKIPPED
[INFO] resource-server .................................... SKIPPED
[INFO] rest ............................................... SKIPPED
[INFO] sw360 .............................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.111 s
[INFO] Finished at: 2020-03-11T14:47:32+01:00
[INFO] Final Memory: 48M/381M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.thrift.tools:maven-thrift-plugin:0.1.11:compile (thrift-sources) on project datahandler: thrift did not exit cleanly. Review output for more information. -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.thrift.tools:maven-thrift-plugin:0.1.11:compile (thrift-sources) on project datahandler: thrift did not exit cleanly. Review output for more information.
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:863)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:199)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoFailureException: thrift did not exit cleanly. Review output for more information.
        at org.apache.thrift.maven.AbstractThriftMojo.execute(AbstractThriftMojo.java:177)
        at org.apache.thrift.maven.ThriftCompileMojo.execute(ThriftCompileMojo.java:21)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
        ... 20 more
[ERROR]
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :datahandler
dira@server:/usr/local/src/sw360$
```

**Comment by mcjaeger** [comment]
Hello @dineshr93, thanks for actually clarifying it. From the output that you have provided:
```
dira@server:/usr/local/src/sw360$ thrift --version
Thrift version 0.9.1
```
I guess the thrift version is not OK, you chould use 0.11. you could use also 0.12. I am not aware if someone tested it ever with 0.13 so far, so 0.11 and 0.12 should be OK:

It is Ok to open issues if you exprience problems, but maybe .... may I ask to have the patience and have a look at least to the project readme first? It is at 

* https://github.com/eclipse/sw360/blob/master/README.md

You could also have a look at (wiki home page)

* https://github.com/eclipse/sw360/wiki#frequently-experienced-problems

Please feel free to open issues in case you see problems, but also from my side, I am not sure how I can make more clear in documentation that thrift 0.9.1 will not work. I guess the answer is to provide packages for sww360, right?

**Comment by mcjaeger** [comment]
I see now that the first line in the link that you have provided tells you that thrift 0.11 is required, are you sure you have used that link?

<img width="926" alt="Screenshot 2020-03-11 at 16 52 07" src="https://user-images.githubusercontent.com/3418977/76436624-bf218a00-63b8-11ea-87e2-885a3e3ab2ac.png">

**Comment by dineshr93** [comment]
Yes, I have read them. For those who tries to install in Ubuntu 16.04, 0.9.1 is the default install of `sudo apt get install thrift-compiler` command mentioned in that wiki. They should refrain from installing thrift via command on non Ubuntu 19. 

Now I used `install-thrift.sh` script to install `0.11.0` finally.

Build got successful & webapp is up 

I ran into another problem though. when I try to access the webapp from another computer (as my team should access the sw360 deployed in server)

> https://www.sw360-company-server.com:8443

I can see the web page as well

But now all clickable option gets redirects to 

> localhost

for example clicking `sign in` redirects to `https://localhost:8443/c/portal/login?p_l_id=2`
instead of 

> https://sw360-company-server.com:8443/c/portal/login?p_l_id=2

![image](https://user-images.githubusercontent.com/4498415/76479167-f7fd4580-6430-11ea-9839-a7a7440d5817.png)



I am sorry if its stupid question again.
